### PR TITLE
add sidepanel and tabbed page #3830

### LINKED ISSUE #3830: [FEATURE] Split resources page into multiple sub pages
The goal of this issue is to split the overloaded [Resources](https://ghostfol.io/en/resources) page into [tabbed sub pages](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/about/about-page.component.ts#L20)

* `en/resources` - _Ghostfolio_
* `en/resources/guides` - _Guides_
* `en/resources/markets` - _Markets_
* `en/resources/glossary` - _Glossary_

The [About](https://github.com/ghostfolio/ghostfolio/tree/main/apps/client/src/app/pages/about) page can be used as a reference.

### PR DESCRIPTION
**Pull request by STSonyThomas**
closes #3830

### CODE REVIEW

File: .vscode/settings.json
Comment (dtslvr) [inline .vscode/settings.json:1]: Please revert this change

File: package.json
Comment (dtslvr) [inline package.json:1]: Please revert this change

File: package-lock.json
Comment (dtslvr) [inline package-lock.json:1]: Please revert this change

File: apps/client/src/app/pages/resources/resources-page.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/resources-page.scss:1]: Please revert unneeded styles

File: apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss:1]: Please revert unneeded styles

File: apps/client/src/app/pages/resources/glossary/resources-glossary.component.ts [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'gf-resources-glossary',
+  templateUrl: './resources-glossary.component.html',
```
Comment (dtslvr) [inline apps/client/src/app/pages/resources/glossary/resources-glossary.component.ts:5]: Move down to order alphabetically

File: apps/client/src/app/pages/resources/guides/resources-guides.component.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/guides/resources-guides.component.scss:1]: Please revert unneeded styles

File: apps/client/src/app/pages/resources/markets/resources-markets.component.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/markets/resources-markets.component.scss:1]: Please revert unneeded styles

File: apps/client/src/app/pages/resources/markets/resources-markets.component.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
+import { Component } from '@angular/core';
+
+@Component({
+  selector: 'gf-resources-markets',
+  templateUrl: './resources-markets.component.html',
```
Comment (dtslvr) [inline apps/client/src/app/pages/resources/markets/resources-markets.component.ts:5]: Please move down to order alphabetically

File: apps/client/src/app/pages/resources/overview/resources-overview.component.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/overview/resources-overview.component.scss:1]: Please revert unneeded styles

File: apps/client/src/app/pages/resources/resources-page-routing.module.ts [hunk 3]
```diff
@@ -3,22 +3,48 @@ import { AuthGuard } from '@ghostfolio/client/core/auth.guard';
 import { NgModule } from '@angular/core';
 import { RouterModule, Routes } from '@angular/router';
 
+// import { ResourcesGlossaryPageComponent } from './glossary/resources-glossary.component';
```
Comment (dtslvr) [inline apps/client/src/app/pages/resources/resources-page-routing.module.ts:6]: Clean up please

File: apps/client/src/app/pages/resources/resources-page.component.ts [hunk 4]
```diff
@@ -6,10 +6,10 @@ import { Component, OnInit } from '@angular/core';
 import { Subject } from 'rxjs';
 
 @Component({
-  host: { class: 'page' },
+  host: { class: 'page has-tabs' },
   selector: 'gf-resources-page',
-  styleUrls: ['./resources-page.scss'],
-  templateUrl: './resources-page.html'
+  templateUrl: './resources-page.html',
```
Comment (dtslvr) [inline apps/client/src/app/pages/resources/resources-page.component.ts:11]: Move down to order alphabetically

File: apps/client/src/app/pages/resources/resources-page.module.ts [hunk 5]
```diff
@@ -1,12 +1,19 @@
...
+import { MatTabsModule } from '@angular/material/tabs';
+import { RouterModule } from '@angular/router';
 
 import { ResourcesPageRoutingModule } from './resources-page-routing.module';
 import { ResourcesPageComponent } from './resources-page.component';
 
 @NgModule({
   declarations: [ResourcesPageComponent],
-  imports: [CommonModule, ResourcesPageRoutingModule],
+  imports: [
```
Comment (dtslvr) [inline apps/client/src/app/pages/resources/resources-page.module.ts:11]: Please order imports alphabetically

File: apps/client/src/app/pages/resources/overview/resources-overview.component.html [hunk 6]
```diff
@@ -0,0 +1,16 @@
+<div class="container">
+  <div class="row">
+    <div class="col">
+      <h1 class="h3 mb-4 text-center" i18n>Resources Overview</h1>
```
Comment (dtslvr) [inline apps/client/src/app/pages/resources/overview/resources-overview.component.html:4]: Resources (remove Overview)

**Review by dtslvr — CHANGES_REQUESTED** [review]
Thanks for your contribution, @STSonyThomas! 😎
I have reviewed your changes and added a few comments.

File: apps/client/src/app/pages/resources/glossary/resources-glossary.component.ts (hunk 1 above)
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/glossary/resources-glossary.component.ts:5]: this is what you meant right?
image

File: apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss:1]: I only see the scroll bar style as an unneeded style. Could you let me know which other you would want me to remove?

File: apps/client/src/app/pages/resources/overview/resources-overview.component.html (hunk 6 above)
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/overview/resources-overview.component.html:4]: So should I render one of the markets, Glossary or Guides page on initial render?

File: apps/client/src/app/pages/resources/resources-page.module.ts (hunk 5 above)
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/resources-page.module.ts:11]: image
Is this alright?

File: apps/client/src/app/pages/resources/glossary/resources-glossary.component.ts (hunk 1 above)
Comment (dtslvr) [reply apps/client/src/app/pages/resources/glossary/resources-glossary.component.ts:5]: > this is what you meant right?

Yes 👍🏻

File: apps/client/src/app/pages/resources/overview/resources-overview.component.html (hunk 6 above)
Comment (dtslvr) [reply apps/client/src/app/pages/resources/overview/resources-overview.component.html:4]: No, just adapt the text of `h1` to "Resources ~~Overview~~"

File: apps/client/src/app/pages/resources/resources-page.module.ts (hunk 5 above)
Comment (dtslvr) [reply apps/client/src/app/pages/resources/resources-page.module.ts:11]: > Is this alright?

Yes 👍🏻

File: apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss
Comment (dtslvr) [reply apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss:1]: As this is a pure split up task (without introducing new content), I expect that the style will be the same as in the [original page](https://github.com/ghostfolio/ghostfolio/blob/main/apps/client/src/app/pages/resources/resources-page.scss). All other styles for e.g. `.mb-4` are coming automatically, you don't have to define them yourself.

Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss:1]: Oh got it!

**Comment by STSonyThomas** [comment]
Hey I have made the changes. But then when I tried uninstalling ionic/angular and then comparing with the main branch (the changes had been reflected at package-lock as well) it still showed conflict at package-lock with the main branch. I have a separate branch where ionic/angular has been removed. Do you want me to make a PR with that branch so that you can review it?

**Comment by STSonyThomas** [comment]
Right now there is no conflict with the main branch by the way.

**Comment by dtslvr** [comment]
> Do you want me to make a PR with that branch so that you can review it?

I would try to revert it here, e.g. by copying the original content to your branch ([package.json](https://github.com/ghostfolio/ghostfolio/blob/main/package.json) / [package-lock.json](https://github.com/ghostfolio/ghostfolio/blob/main/package-lock.json))

**Comment by STSonyThomas** [comment]
Yeah I'll do that. I didn't think of that.

**Comment by STSonyThomas** [comment]
Hey I have made those changes as well.

File: apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss:1]: Please use the original style:
```
:host {
  color: rgb(var(--dark-primary-text));
  display: block;

  a {
    color: rgba(var(--palette-primary-500), 1);
    font-weight: 500;

    &:hover {
      color: rgba(var(--palette-primary-300), 1);
    }
  }
}

:host-context(.theme-dark) {
  color: rgb(var(--light-primary-text));
}
```

File: apps/client/src/app/pages/resources/guides/resources-guides.component.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/guides/resources-guides.component.scss:1]: Please use the original style:
```
:host {
  color: rgb(var(--dark-primary-text));
  display: block;

  a {
    color: rgba(var(--palette-primary-500), 1);
    font-weight: 500;

    &:hover {
      color: rgba(var(--palette-primary-300), 1);
    }
  }
}

:host-context(.theme-dark) {
  color: rgb(var(--light-primary-text));
}
```

File: apps/client/src/app/pages/resources/markets/resources-markets.component.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/markets/resources-markets.component.scss:1]: Please use the original style:
```
:host {
  color: rgb(var(--dark-primary-text));
  display: block;

  a {
    color: rgba(var(--palette-primary-500), 1);
    font-weight: 500;

    &:hover {
      color: rgba(var(--palette-primary-300), 1);
    }
  }
}

:host-context(.theme-dark) {
  color: rgb(var(--light-primary-text));
}
```

File: apps/client/src/app/pages/resources/overview/resources-overview.component.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/overview/resources-overview.component.scss:1]: Please use the original style:
```
:host {
  color: rgb(var(--dark-primary-text));
  display: block;

  a {
    color: rgba(var(--palette-primary-500), 1);
    font-weight: 500;

    &:hover {
      color: rgba(var(--palette-primary-300), 1);
    }
  }
}

:host-context(.theme-dark) {
  color: rgb(var(--light-primary-text));
}
```

File: apps/client/src/app/pages/resources/overview/resources-overview.module.ts [hunk 7]
```diff
@@ -0,0 +1,12 @@
+import { CommonModule } from '@angular/common';
+import { NgModule } from '@angular/core';
+import { RouterModule } from '@angular/router';
+
+import { ResourcesOverviewRoutingModule } from './resources-overview-routing.module';
+import { ResourcesOverviewComponent } from './resources-overview.component';
+
+@NgModule({
+  declarations: [ResourcesOverviewComponent],
+  imports: [CommonModule, RouterModule, ResourcesOverviewRoutingModule]
```
Comment (dtslvr) [inline apps/client/src/app/pages/resources/overview/resources-overview.module.ts:10]: `imports: [CommonModule, ResourcesOverviewRoutingModule, RouterModule]`

File: apps/client/src/app/pages/resources/resources-page-routing.module.ts [hunk 8]
```diff
@@ -7,18 +7,40 @@ import { ResourcesPageComponent } from './resources-page.component';
 
 const routes: Routes = [
   {
```
Comment (dtslvr) [inline apps/client/src/app/pages/resources/resources-page-routing.module.ts:9]: Can you please order the attributes alphabetically?

File: apps/client/src/app/pages/resources/shared/resource-page.scss
Comment (dtslvr) [inline apps/client/src/app/pages/resources/shared/resource-page.scss:1]: Please remove

File: apps/client/src/app/pages/resources/resources-page.html [hunk 9]
```diff
@@ -1,258 +1,30 @@
...
+        class="no-min-width px-3"
+        mat-tab-link
+        routerLinkActive
+        [active]="rla.isActive"
+        [routerLink]="tab.path"
+        [routerLinkActiveOptions]="{ exact: true }"
+      >
+        <ion-icon
+          [name]="tab.iconName"
+          [size]="deviceType === 'mobile' ? 'large' : 'small'"
```
Comment (dtslvr) [inline apps/client/src/app/pages/resources/resources-page.html:24]: I think you need to define `deviceType`

(see `this.deviceType = this.deviceService.getDeviceInfo().deviceType;` in other page components)

**Review by dtslvr — CHANGES_REQUESTED** [review]

File: apps/client/src/app/pages/resources/resources-page.html (hunk 9 above)
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/resources-page.html:24]: Hey sorry about that missed that line. I have added it in.

File: apps/client/src/app/pages/resources/shared/resource-page.scss
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/shared/resource-page.scss:1]: I have removed the shared library.

File: apps/client/src/app/pages/resources/resources-page-routing.module.ts (hunk 8 above)
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/resources-page-routing.module.ts:9]: Hey sorry about that I missed those ones. Have handled that as well.

File: apps/client/src/app/pages/resources/overview/resources-overview.module.ts (hunk 7 above)
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/overview/resources-overview.module.ts]: Hey I have made these changes as well.

File: apps/client/src/app/pages/resources/overview/resources-overview.component.scss
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/overview/resources-overview.component.scss:1]: Hey I have reverted to the original.

File: apps/client/src/app/pages/resources/markets/resources-markets.component.scss
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/markets/resources-markets.component.scss:1]: Done.

File: apps/client/src/app/pages/resources/guides/resources-guides.component.scss
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/guides/resources-guides.component.scss:1]: Done.

File: apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss
Comment (STSonyThomas) [reply apps/client/src/app/pages/resources/glossary/resources-glossary.component.scss:1]: Done.

**Comment by STSonyThomas** [comment]
Hey done with the mentioned changes as well.

**Review by dtslvr — APPROVED** [review]
Thanks for your contribution, @STSonyThomas! 😎

**Comment by STSonyThomas** [comment]
Lessgooo! Also you didn't forget to merge this right? You usually merge in batches right? :)
